minard/standard_runs.py

```diff
@@ -0,0 +1,50 @@
+import couchdb
+from itertools import groupby
+from .views import app
+from time import time
+from uuid import uuid4
+
+def get_standard_runs():
+    couch = couchdb.Server("https://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    orca_db = couch['orca']
+    sr_view = orca_db.view("standardRuns/getStandardRunsWithVersion")
+    # standard run keys are [doc-version, run name, run version, timestamp]
+    # I want to groupby by run name then within that group all run versions sorted
+    # by timestamp
+    rows = sorted(sr_view.rows, key=lambda x: x.key[1] + x.key[2])
+    groups = groupby(rows, lambda x: x.key[1] + x.key[2])
+    groups = [(x, list(y)) for x, y in groups]
+    runs = [max(group, key=lambda x: x.key[3]) for _, group in groups]
+    runs = sorted(runs, key=lambda x: x.key[1])
+    runs = [(x, list(y)) for x,y in groupby(runs, lambda x: x.key[1])]
+    return runs
+
+def get_standard_run(uuid):
+    couch = couchdb.Server("https://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    orca_db = couch['orca']
+    return orca_db.get(uuid)
+
+def update_standard_run(uuid, new_values):
+    couch = couchdb.Server("https://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    orca_db = couch['orca']
+    doc = dict(orca_db.get(uuid))
+    for k, v in new_values.iteritems():
+        doc[k] = v
+    doc["_id"] = uuid4().hex
```

Comment:
I just checked how the standard runs are saved in Orca and they don't explicitly include an id version. @mastbaum or @jcaravaca can you confirm that this is kosher?

Reply:
As far as I know a document always has an id field, if it doesn't couch creates one for a document. The python couchdb library recommends the client create the hash though b/c if the libraries POST request fails and retries you can end up posting the same document multiple times.

Reply:
Ah ok that makes sense.